spec/controllers/admin/tracks_controller_spec.rb

```diff
@@ -409,31 +406,80 @@
     end
 
     context 'has start_date and end_date' do
-      it_behaves_like 'fails to accept', true, true, false
+      before :each do
+        self_organized_track.start_date = Date.current
+        self_organized_track.end_date = Date.current
+        self_organized_track.room = nil
+        self_organized_track.save!
+      end
+
+      it_behaves_like 'fails to accept'
     end
 
     context 'has start_date and room' do
-      it_behaves_like 'fails to accept', true, false, true
+      before :each do
+        self_organized_track.start_date = Date.current
+        self_organized_track.end_date = nil
+        self_organized_track.room = @room
+        self_organized_track.save!
+      end
+
+      it_behaves_like 'fails to accept'
     end
 
     context 'has start_date' do
-      it_behaves_like 'fails to accept', true, false, false
+      before :each do
+        self_organized_track.start_date = Date.current
+        self_organized_track.end_date = nil
+        self_organized_track.room = nil
+        self_organized_track.save!
+      end
+
+      it_behaves_like 'fails to accept'
     end
 
     context 'has end_date and room' do
-      it_behaves_like 'fails to accept', false, true, true
+      before :each do
+        self_organized_track.start_date = nil
+        self_organized_track.end_date = Date.current
+        self_organized_track.room = @room
+        self_organized_track.save!
+      end
+
+      it_behaves_like 'fails to accept'
     end
 
     context 'has end_date' do
-      it_behaves_like 'fails to accept', false, true, false
+      before :each do
+        self_organized_track.start_date = nil
+        self_organized_track.end_date = Date.current
+        self_organized_track.room = nil
+        self_organized_track.save!
+      end
+
+      it_behaves_like 'fails to accept'
     end
 
     context 'has room' do
-      it_behaves_like 'fails to accept', false, false, true
+      before :each do
+        self_organized_track.start_date = nil
+        self_organized_track.end_date = nil
+        self_organized_track.room = @room
+        self_organized_track.save!
+      end
+
+      it_behaves_like 'fails to accept'
     end
 
-    context 'has non of start_date, end_date, room' do
-      it_behaves_like 'fails to accept', false, false, false
+    context 'has none of start_date, end_date, room' do
+      before :each do
+        self_organized_track.start_date = nil
```

Comment:
We can have another variable for a track that has none of those extra attributes (dates, room) set. The same variable can be used in other cases, where you only want to set one of the attribute to non nil.